packages/babel-generator/src/buffer.js

```diff
@@ -14,6 +14,7 @@ export default class Buffer {
     this._indent = format.indent.base;
     this.format = format;
     this.buf = "";
+    this.last = "";
```

Comment:
There will be a performance issue if we do that.

v8 stores strings created from concatenation as a [ConsString](https://github.com/v8/v8/blob/master/src/objects.h#L9042). And when we access any character of it, v8 has to perform a [flatten operation](https://github.com/v8/v8/blob/master/src/x64/codegen-x64.cc#L424), that will cost a bunch of time in our case.

Here is a simple node app for testing such a feature.

```javascript
var code = require('fs').readFileSync('jquery.js', 'utf-8');

var step = 200;
var pieces = [];
for (var i = 0, ii = code.length; i < ii; i += step) {
  pieces.push(code.substring(i, step));
}

var buf = '';
var ts = Date.now();
for (var i = pieces.length - 1; i >= 0; i--) {
  buf += pieces[i];
}
console.log('concat', Date.now() - ts);

ts = Date.now();
buf[0]; // flatten here
console.log('read first', Date.now() - ts);

ts = Date.now();
buf[buf.length - 1];
console.log('read last', Date.now() - ts);
```

And here is the my result.

```
concat 8
read first 100
read last 0
```